edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/models/DockerStats.cs

```diff
@@ -4,98 +4,170 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Docker.Models
 {
     using System;
     using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Util;
     using Newtonsoft.Json;
 
     public class DockerStats
     {
+        [JsonConstructor]
+        public DockerStats(string name, Dictionary<string, DiskIO[]> block_io_stats, DockerCpuStats cpu_stats, MemoryStats memory_stats, Dictionary<string, NetworkInfo> networks, int? num_processes, PidsStats pids_stats, DateTime? read)
+        {
+            this.Name = Option.Maybe(name);
```

Comment:
what is this name used for? should it be required?

Reply:
It is required by the stats that use this, but this class doesn't and shouldn't know that. All this does is parse the docker api result. The class that uses this handles the empty name case